src/6_branch.js

```diff
@@ -67,6 +67,10 @@ var wrap = function(parameters, func, init) {
 			 * @type {function(?Error,?): undefined}
 			 */
 			var done = function(err, data) {
+				if (err) {
```

Comment:
As for putting the reset within the next if statement, what I saw is that some of the tests fired off asserts in callbacks, so the same thing happened.